### DISCUSSION TITLE
Possible to configure frigate to only listen on a specific interface?

### DISCUSSION THREAD

**Discussion by hagak** [blakeblackshear/frigate#4182] [Ask A Question]
I have frigate setup such that it has 2 interfaces, 1 interface is where my cameras live so it needs this interface to actually connect to the cameras.

Another interface is where the API and all service requests to frigate happen.  I would like Frigate to ONLY listen on this interface, not on the camera interface.  While the camera network is limited I see no reason to expose a service with no auth. on that interface.

**Comment by NickM-27** [comment]
Can you elaborate more on what you mean by interface? Are you using a reverse proxy or something else to set this up?

**Reply by hagak** [reply]
Multiple network adapters, each on a different network.  Frigate services appear to be listening on ALL interfaces, it would be nice to be able to tell Frigate to only use a specific interface, usually you would so something like this:
Interface1: 192.168.10.5
Interface2: 10.10.10.5

Frigate listen: 10.10.10.5:5000. (this would then only respond to requests that come on Interface2). Currently Frigate will respond to requests on ANY interface.

**Reply by NickM-27** [reply]
Thanks for the extra detail. This is something that would be [setup in docker](https://docs.docker.com/network/iptables/) or setup inside of nginx or another reverse proxy.

**Reply by hagak** [reply]
While you can accomplish most of this within Docker with reverse proxies (that is what I am doing to put auth in place). BUT I have a macvlan connected to this docker image to give it access to my Camera Network.  This will prevent or make very difficult any of the docker mechanisms.

**Reply by NickM-27** [reply]
https://github.com/blakeblackshear/frigate/issues/4144 may be of interesting but to answer the question, this is not supported in frigate directly

**Comment by natevw** [comment]
It looks like this might soon-ish become possible! Via https://github.com/blakeblackshear/frigate/blob/b848c90f026b0e56ed2ba76f128dbb708b41f163/docs/docs/configuration/advanced/system.md#listen-on-different-ports found https://github.com/blakeblackshear/frigate/pull/21779 which has been merged into the 0.18 branch, thank you! Wanting it for similar reasons as many have mentioned: to avoid excess traffic load on my home router, I want/need my Frigate container connected directly to my cameras VLAN so Frigate can open connections to them — but do **not** want said cameras VLAN to be opening connections to Frigate.
